### ISSUE TITLE
Asking 

### ISSUE THREAD

**Issue by sopheap77** [PENGZhaoqing/SimpleSniff#1]
Hello.
I just want to you something regarding to this issue. For my case, I only want to read pcap file. By clicking open pcap file, all packets will be listed in TableWidget. Could you please explain me about this problem?

**Comment by PENGZhaoqing** [comment]
What do you want me to explain?  TableWidget is a tool to display(or print) something, and you should first parse the packets with pcap and then convert to string, then you can display them with TableWidget

**Comment by sopheap77** [comment]
Hello Sir,
Thank you for your answer and I understand your answer as well.
Since I am new to pyqt, I might have a lot of questions regarding to this
problem. For your case, you sniff and show the pcap packet onto
tablewidget. For my case, I just want to open the pcap file and display
packets like you did on tablewidget. So, could you please tell me how to
convert pcap file to string?

Sorry for the inconveniences. Lookinh foward to seeing your response.

Sincerely yours,
KEY Sopheap

**Comment by sopheap77** [comment]
Hello sir,
In your case, you display source, destination, protocol and more by
str(source), str(destination), str(protocol). So, after I parse pcap
packet, I should do the same. Am i right?

**Comment by PENGZhaoqing** [comment]
ok, I will give you some hints.

 `self.filterButton.clicked.connect(self.start_thread)` says that the start_thread function will be called after I clicked on filter button

`text = str(self.filterText.text())` says that the I will get the input text from filter widget

`self.thread = SniffThread(self, self.pkt_buffer)` says that I initialize a thread (not started yet)

`self.connect(self.thread, SIGNAL("output(PyQt_PyObject)"), self.push_entry)` means that, under this thread, if I send something(e.g. l) with signal  "output(PyQt_PyObject)" then the function push_entry will handle this l (in my example, push_entry will print this l).

`self.thread.render(interface=interface, filter_text=text)` says that I will start the thread now (the sniff will start).

Actually, pyqt is not that complicated if you just want to display something without thread, you can refer some tutorial:

https://pythonspot.com/pyqt4/

http://zetcode.com/gui/pyqt4/firstprograms/

But in my case, since a sniff operation is a thread, so I make it become so complicated.

**Comment by sopheap77** [comment]
Thank you for your hints. I am really appreciate it.
By the way, one more thing about print_summary function. Does that part use
to convert pcap packets to string and seperate each string into each column
right?

**Comment by PENGZhaoqing** [comment]
yes, print_summary is callback function that handle the result from sniff function, it convert the packet result into lists

**Comment by sopheap77** [comment]
Okay, I understand it.
Another thing, the variable pkt in the print_summary function, is it the
pcap that you sniff?
For my case, I just open the pcap file from my directory and use expand
function to my imported pcap file. Since I don’t create a thread, that emit
signal is not important. Am I right?

**Comment by sopheap77** [comment]
Hello Sir,
About the parsing pcap file and display on TableWidget, now I can display
on TableWidget. But why it only display the last packet of pcap not all
packet? Could you tell me the problem here?

Sincerely yours,
KEY Sopheap

**Comment by sopheap77** [comment]
Hello Sir,

Here is my result. Hope you can help me one last time.

Sincerely yours,
KEY Sopheap
